Define addtag to copy the source register if MT is disabled (#98)
riscv/riscv-memory-tagging: src/mte_tag.adoc

places incremented `pointer_tag` value in `rd[XLEN:XLEN-pointer_tag_width+1]`.

If memory tagging is disabled in the current execution environment (see

> <<MEM_TAG_EN>>), then `addtag` instruction copies `rs1` into `rd.

Reviewer: nit: s/copies/moves

], config:{lanes: 1, hspace:1024}}
....

[NOTE]
=====
`addtag`, unlike the other instructions that are taken from Zimops codepoints,
has three different semantics depending on execution environment

* MT is not implemented: `rd` is zeroed;

> * MT is implemented by disabled in the current execution environment: `rs1`

Reviewer 2: ```suggestion
* MT is implemented but disabled in the current execution environment: `rs1`
```

  is copied into `rd`;
* MT is enabled in the current execution environment: the tag addition is
  performed.

> Because that menas `rd` always contains a valid pointer if MT is implemented,

Reviewer 2: ```suggestion
That means `rd` always contains a valid pointer if MT is implemented,
```

        addi s2, sp, 32
        settag s2, 1
-----

==== Example: optimized stack tagging codegen

If we know we are compiling for a target that implements Zimt, we can
optimize the stack codegen to skip `or` when constructing pointers to

> objects.

Reviewer 2: ```suggestion
Software that targets implementations which support Zimt can
optimize the stack codegen to skip `or` when constructing pointers to
objects.
```
